### ISSUE TITLE
Fix Style/RedundantSelf no keywords parameter issue

### ISSUE THREAD

**Issue by bkuhlmann** [rubocop/rubocop#13525]
## Expected behavior

Prior to the release of RuboCop 1.69.0, the following caused no RuboCop issues when defining a method that _would not accept_ any keyword arguments:

``` ruby
Module.new { def test(**nil) = super }
```

## Actual behavior

After the release of RuboCop 1.69.0, the following error arrises when using the above code:

``` 
1 error occurred:
An error occurred while Style/RedundantSelf cop was inspecting /Users/bkuhlmann/Engineering/Misc/demo:23:21.
```

:warning: I'm not able to provide `rubocop --debug` details because GitHub has a character limit error when submitting this issue. I can provide this information via a Gist if that would help?

## Steps to reproduce the problem

Ensure `Style/RedundantSelf` is enabled and run RuboCop against the above code snippet.

## RuboCop version

``` 
1.69.0 (using Parser 3.3.6.0, rubocop-ast 1.36.2, analyzing as Ruby 3.3, running on ruby 3.3.6) [arm64-darwin24.1.0]
  - rubocop-capybara 2.21.0
  - rubocop-packaging 0.5.2
  - rubocop-performance 1.23.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 3.2.0
  - rubocop-thread_safety 0.6.0
```

**Comment by viralpraxis** [comment]
It should be already fixed by https://github.com/rubocop/rubocop/pull/13521
